edu-services/gradebook-service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/GradingEvent.hbm.xml

```diff
@@ -20,6 +20,7 @@
 		<property name="studentId" column="STUDENT_ID" type="string" not-null="true"/>
 		<property name="dateGraded" column="DATE_GRADED" type="timestamp" not-null="true"/>
 		<property name="grade" column="GRADE" type="string" />
+		<property name="excluded" column="IS_EXCLUDED" type="boolean" not-null="false"/>
```

Comment:
As with the reference PR, I believe this column should be not-nullable. Conceptually, there is only two states: grade is included, or grade is excluded. Eliminating the 3rd state (null) makes this feature cleaner, easier to understand, and better conforms to standard database design practises.

Reply:
Hello, @bjones86, we've been testing this PR and we would need a null value because you can modify a grade but not modifying the "excluding" state so it's needed to be null because on the controller excluded value is generated null because is not modified in the event.

Null is threated as it on line 1870 of GradebookServiceHibernateImpl. You can see it too GradeLogPanel.java, on line 79. This is what it was designed. If we need to remove null this should be entirely refactored.

What do you think?

Reply:
Personally I feel that this should be refactored to remove null as a possible value. I don't think there is a use case for having these as nullable. What does null represent? Conceptually, the grade is either included (false) or excluded (true).

With regards to `GradebookServiceHibernateImpl:1870`:

```
public void saveGradeAndExcuseForStudent(final String gradebookUid, final Long gradableObjectId, final String studentUid,
			final String grade, final boolean excuse) {
```

In this case it is a primitive `boolean`, which cannot be null. Primitive booleans must be initialized as either `true` or `false`, so I don't think it's possible to have a null value here.

Long story short, I think the code should be refactored to eliminate null as a possible value here, as it doesn't provide any value. I'm sorry if this causes you more work  :(

If anyone disagrees with me, or has another opinion, feel free to chime in.

Reply:
@bjones86 , the way it works is not exactly this way, sorry. 

There are, right now, three scenarios in the UI (before this PR). You can exclude a grade, you can include a grade but you can also modify a grade **without changing the state** of included/excluded. In all this three scenarios the event is always the same. We are just mapping the UI behaviour to save consistent events. 

Right now, in the UI, you can do those three scenarios and you'll have the same event for all three. We are trying to avoid that (is weird for the instructor).

Maybe we can talk if the approach is the best way to do it but, for now, there are three cases. Null would be "not changed". 

Should we map it to an integer? Thanks.